packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/useUploadFileEntities.ts

```diff
@@ -375,28 +394,20 @@ export function useUploadFileEntities(
   ])
 
   const uploadProgress: FileUploadProgress[] = useMemo(() => {
-    return [...trackedUploadProgress].map(([file, progress]) => {
+    return [...trackedUploadProgress].map(([file, trackedProgress]) => {
       return {
         file: file,
-        progress: progress.progress,
-        status: progress.status,
-        failureReason: progress.failureReason,
+        progress: trackedProgress.progress,
+        status: trackedProgress.status,
+        failureReason: trackedProgress.failureReason,
         cancel: () => {
           cancelUpload(file)
         },
         pause: () => {
           pauseUpload(file)
         },
         resume: () => {
-          prepareUpload(
-            { files: [file], parentId: progress.parentId },
-            {
-              onSuccess: result => {
-                // the file was already checked before it was paused, so we can skipChecks
-                postPrepareUpload(result, true)
-              },
-            },
-          )
+          startUpload(trackedProgress.filePreparedForUpload)
```

Comment:
since we now store the `filePreparedForUpload` in the TrackedUploadProgress object, we can just call `startUpload` on resume, and remove the `skipChecks` parameter from `postPrepareUpload`